examples/pytorch/graphsage/experimental/train_dist.py

```diff
@@ -165,7 +165,7 @@ def pad_data(nids):
     the maximum size among all workers.
     """
     import torch.distributed as dist
-    num_nodes = th.tensor(nids.numel())
+    num_nodes = th.tensor(nids.numel()).to(device)
```

Comment:
I don't think num_nodes need to be on GPU? Did you meet any error here?

Reply:
Yes, if the num_nodes is not on GPU and use 'nccl' backend, we'll get the following error:
```
  File "train_dist_dgs.py", line 190, in pad_data
    dist.all_reduce(num_nodes, dist.ReduceOp.MAX)
  File "/usr/local/lib/python3.6/dist-packages/torch/distributed/distributed_c10d.py", line 1285, in all_reduce
    work = default_pg.allreduce([tensor], opts)
RuntimeError: Tensors must be CUDA and dense
```

Reply:
I see, nccl only supports GPU tensor